### DISCUSSION TITLE
Understanding the best way to backtrack my project while still keeping what I have

### DISCUSSION THREAD

**Discussion by strikeouts27** [community/community#106567] [Programming Help]
### Body

I have a project in a  repository that has an earlier version of my project that I want to work on. Basically a django mentor of mine said that working from a template would be a good idea. However i have not been able to get it off the ground. I am wanting to preserve what I have currently in my project, while at the same time doing a git reset to the earlier point in time. Basically I want a repository for the earlier point in time and what I have now. Is there a way to do this? I have tried uploading repositories into github and when I do so I get the repository in a repository and that made my whole repository unusable and I have lost work doing that. 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by spenserblack** [comment]
Are you able to identify the exact commit that you would like to return to? There are several ways to do this, I'll add a couple:
```shell
# These two should be roughly the same, it just comes down to preference

# create a branch from the earlier point then checkout
git branch second-try <COMMIT>
git checkout second-try

# checkout earlier point then branch from it. You might want to do this
# to ensure that you're branching off of your project in the state that
# you want it to be first.
git checkout <COMMIT>
git checkout -b second-try
```

In both of these `<COMMIT>` will be some identifier for the earlier point. If you already know the exact commit, you can use the commit's hash. For example, `git checkout 8d61ecb7d5705ab42f064eb7921b29897527f886` to return to https://github.com/strikeouts27/djangodelights/commit/8d61ecb7d5705ab42f064eb7921b29897527f886. If you know it's, for example, 20 commits back, then `<COMMIT>` can be `HEAD~20`. If you don't know the commit you want to reset to, you can use a `git bisect` to find it, but that's a bit of an advanced topic for a git beginner :slightly_smiling_face:

You might have noticed that the examples I gave you don't rewrite the history of `main`, but instead create a new branch for your second try. While you *can* rewrite the history of main, and put your old/first try on a different branch instead, creating a new branch for your "second try" is going to be a bit safer than rewriting history, especially if you're just getting started using git. Once you get to a good point you can always just rename the branches.
